lib/lua/lmathlib.c

```diff
@@ -181,6 +181,7 @@ static int math_max (lua_State *L) {
 static int math_random (lua_State *L) {
   /* the `%' avoids the (rare) case of r==1, and is needed also because on
      some systems (SunOS!) `rand()' may return a value larger than RAND_MAX */
+    // DISCUSSME: what do we do here?
   lua_Number r = (lua_Number)(rand()%RAND_MAX) / (lua_Number)RAND_MAX;
```

Comment:
Replace with a call to `rand32f()`, maybe?

Reply:
Maybe leave as-is and translate the existing logic to use `next()` and `MAX_VALUE` actually. The cast to `lua_Number` before the division here is actually significant — `lua_Number` is a typedef for `double` and so the calculation is being performed with greater precision.

Reply:
This is external code. It should not be changed.